### ISSUE TITLE
Using validate(this) in init occurs error when it comes to open / abstract class 

### ISSUE THREAD

**Issue by JayAhn2** [valiktor/valiktor#69]
## Expected Behavior
Probably this report is due to my lack of proficiency in Kotlin. I want to validate values of a constructor in abstract or open class. If there is no way to achieve it as below. Please guide me how to do the same thing. 

```kotlin
abstract class Quantity(open val value: Int) : BaseValueObject() {
    init {
        validate(this) {
            validate(Quantity::value).hasDigits()
        }
    }
}
```

## Actual Behavior
![image](https://user-images.githubusercontent.com/25868002/131414029-2bf08f47-3a66-445d-9aa9-aa37bde5438b.png)

## Steps to Reproduce the Problem

  1. Write abstract class as above
## Example Code or Link to a Project
As above

## Environment

  - Valiktor version: 0.12.0
  - JDK/JRE version: Open JDK 11
  - Platform/OS: Mac

**Comment by PierrickPuimeanChieze** [comment]
Yeah, logical here. Reason given here : https://stackoverflow.com/questions/9851813/java-leaking-this-in-constructor/9851843#9851843. Nothing valiktor could correct.

One suggestion, maybe : write a method instead, for example executeValidation(), and call this method from constructor or whatever in the child classes. This is the approach I elected to follow.
